src/Form/SpecialBenefitType.php

```diff
@@ -17,9 +18,9 @@ final class SpecialBenefitType extends AbstractType implements DataMapperInterfa
 {
     private $currency;
 
-    public function __construct(string $currency)
+    public function __construct(Currency $currency)
     {
-        $this->currency = $currency;
+        $this->currency = $currency->getValue();
```

Comment:
Leave it as a Currency object and call the getValue method inside of the buildForm method.

Reply:
No resolved yet